packages/utils/src/common/array-intersection.ts

```diff
@@ -0,0 +1,17 @@
+type ArrayIntersectionElement = string | number
+
+export function arrayIntersection<TElement = ArrayIntersectionElement>(
+  firstArray: TElement[],
+  secondArray: TElement[]
+): TElement[] {
+  const firstArraySet = new Set(firstArray)
+  const res = new Set<TElement>()
+
+  secondArray.forEach((element) => {
+    if (firstArraySet.has(element)) {
+      res.add(element)
+    }
+  })
+
+  return Array.from(res)
```

Comment:
I think you can get a union through spreading 2 sets

```suggestion
  const firstArraySet = new Set(firstArray)
  const secondArraySet = new Set(secondArray)
  
  return Array.from(new Set([...firstArraySet, ...secondArraySet]););
```